Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -52,8 +64,12 @@ int lightsCheck (uint8_t testLight)
 {
 	uint8_t maskedHeadLights = testLight & 0b00000111;
 	uint8_t maskedSignals = testLight & 0b00011000;
+	uint8_t maskedHazards = testLight & 0b00111000;
+	uint8_t hazardsOn = testLight & 0b00100000;
+	int oneHeadlightOn = maskedHeadLights && !(maskedHeadLights & (maskedHeadLights-1));
+	int oneSignalOn = maskedSignals && !(maskedSignals & (maskedSignals-1));
 	
-	if(!(maskedHeadLights && !(maskedHeadLights & (maskedHeadLights-1))) || !(maskedSignals && !(maskedSignals & (maskedSignals-1)))) // if more than 1 headlight state is chosen or more than one signal is on return 0
+	if(!oneHeadlightOn || (hazardsOn && maskedHazards != 0b00111000) || (!hazardsOn && !oneSignalOn)) // if more than 1 headlight state is chosen or both signals are not on while hazard is on or more than one signal is on return 0
```

Comment:
If none of the signals lights are on, you'll return a 0, but it's valid for neither of the signal lights to be on